iModelCore/ecobjects/src/SchemaMerger.cpp

```diff
@@ -277,6 +317,13 @@ BentleyStatus SchemaMerger::MergeSchemas(SchemaMergeResult& result, bvector<ECSc
     if(failedToFillSchemas)
         return BentleyStatus::ERROR;
 
+     if(SchemaMerger::ValidateUniqueSchemaNames(result.GetResults()) != BentleyStatus::SUCCESS)

```

Comment:
I do not think this should ever happen, or can you hit it with a test?

Reply:
I could, but from what [this comment](https://github.com/iTwin/imodel-native/pull/644#discussion_r1517532994) suggests that was the cause of a bug.